tests/Fixtures/SubscriptionRepositorySpy.php

```diff
@@ -26,4 +26,14 @@ public function getSubscriptions(): array {
 		return $this->subscriptions;
 	}
 
+	public function findByConfirmationCode( string $confirmationCode )
+	{
+		foreach( $this->subscriptions as $subscription ) {
+			if ( $subscription->getConfirmationCode() === $confirmationCode ) {
+				return $subscription;
+			}
+		}
```

Comment:
Nothing against having a `InMemorySubscriptionRepository` (which does not need full implementation, but would need to throw exceptions for stuff not implemented), it should just not be done in the spy.